tiled/authenticators.py

```diff
@@ -219,6 +220,54 @@ async def authenticate(self, request: Request) -> Optional[UserSessionState]:
         return UserSessionState(verified_body["sub"], {})
 
 
+class ProxiedOIDCAuthenticator(OIDCAuthenticator):
+    configuration_schema = """
+$schema": http://json-schema.org/draft-07/schema#
+type: object
+additionalProperties: false
+properties:
+  audience:
+    type: string
+  client_id:
+    type: string
+  well_known_uri:
+    type: string
+  confirmation_message:
+    type: string
+  use_external_authorization
+    type: bool
+"""
+
+    def __init__(
+        self,
+        audience: str,
+        client_id: str,
+        well_known_uri: str,
+        scopes: list[str],
+        confirmation_message: str = "",
+    ):
+        super().__init__(
+            audience=audience,
+            client_id=client_id,
+            client_secret="",
+            well_known_uri=well_known_uri,
+            confirmation_message=confirmation_message,
+        )
+        self.scopes = scopes
+        self._oidc_bearer = OAuth2AuthorizationCodeBearer(
+            authorizationUrl=str(self.authorization_endpoint),
+            tokenUrl=self.token_endpoint,
+        )
+
+    @functools.cached_property
+    def keys(self):
+        return httpx.get(self.jwks_uri).raise_for_status().json().get("keys", [])
```

Comment:
This should have a time-to-live or other way of the cache being invalidated, as the oauth server should be rotating keys periodically.

Reply:
This cache also isn't being used in the superclass authenticate method, it's calling it each time it authenticates